### PR TITLE
Feature/ds4 33 figure out how to autogenerate soap client

### PR DESCRIPTION
**Pull request by GustavFrandsen**
så er pom.xml klar til at generer en soap client og går ud fra alle skal køre clean install på den fordi at den behandler clienten som kildekode og dermed ikke rigtigt er synligt i modulet men man kan finde hvis man åbner stifinder osv. og dermed bliver de heller ikke commitet med i github desktop. kan dog godt være der mangler lidt dependecy men det finder jeg ud af når jeg laver warehouse.

### CODE REVIEW

File: backend/SOAP Client/src/main/java/com/example/soapclient/SoapClientApplication.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class SoapClientApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(SoapClientApplication.class, args);
+    }
+
+}
```
Comment (MagnusHJensen) [inline backend/SOAP Client/src/main/java/com/example/soapclient/SoapClientApplication.java:13]: We don't need the application inside here, but we need a configuration class that creates a `@Bean` definition for the `IEmulatorService`

File: backend/SOAP Client/pom.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            <version>1.7.32</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.cxf</groupId>
+                <artifactId>cxf-codegen-plugin</artifactId>
+                <version>3.5.5</version>
```
Comment (MagnusHJensen) [inline backend/SOAP Client/pom.xml:76]: Can be bumped to `4.0.0`

File: backend/SOAP Client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        <dependency>
+            <groupId>org.glassfish.jaxb</groupId>
+            <artifactId>jaxb-runtime</artifactId>
+            <version>2.3.1</version>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.2.3</version>
+        </dependency>
```
Comment (MagnusHJensen) [inline backend/SOAP Client/pom.xml:43]: Should be replaced with 4.0.0 deps:
```xml
<dependency>
            <groupId>jakarta.xml.ws</groupId>
            <artifactId>jakarta.xml.ws-api</artifactId>
            <version>4.0.0</version>
        </dependency>
        <dependency>
            <groupId>jakarta.annotation</groupId>
            <artifactId>jakarta.annotation-api</artifactId>
            <version>2.1.0</version>
        </dependency>
        <dependency>
            <groupId>com.sun.xml.ws</groupId>
            <artifactId>rt</artifactId>
            <version>4.0.1</version>
</dependency>
```

File: backend/SOAP Client/pom.xml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-devtools</artifactId>
+            <scope>runtime</scope>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>javax.jws</groupId>
+            <artifactId>javax.jws-api</artifactId>
+            <version>1.1</version>
+        </dependency>
```
Comment (MagnusHJensen) [inline backend/SOAP Client/pom.xml:54]:
```suggestion
```

File: backend/SOAP Client/pom.xml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.2.3</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-devtools</artifactId>
+            <scope>runtime</scope>
+            <optional>true</optional>
+        </dependency>
```
Comment (MagnusHJensen) [inline backend/SOAP Client/pom.xml:49]:
```suggestion
```

File: backend/SOAP Client/pom.xml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
```
Comment (MagnusHJensen) [inline backend/SOAP Client/pom.xml:63]:
```suggestion
```

File: backend/SOAP Client/pom.xml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.32</version>
+        </dependency>
+    </dependencies>
```
Comment (MagnusHJensen) [inline backend/SOAP Client/pom.xml:69]:
```suggestion
```

**Comment by MagnusHJensen** [comment]
Maybe this package should also contain a README.md, stating that the `compile` lifecycle needs to be ran before the code is available.

File: backend/SOAP Client/pom.xml (hunk 3 above)
Comment (GustavFrandsen) [reply backend/SOAP Client/pom.xml:43]: doesnt like when i change them to version 4.0.0

Comment (GustavFrandsen) [reply backend/SOAP Client/pom.xml:43]: and not sure what u mean with this `<dependency>
            <groupId>jakarta.xml.ws</groupId>
            <artifactId>jakarta.xml.ws-api</artifactId>
            <version>4.0.0</version>
        </dependency>
        <dependency>
            <groupId>jakarta.annotation</groupId>
            <artifactId>jakarta.annotation-api</artifactId>
            <version>2.1.0</version>
        </dependency>
        <dependency>
            <groupId>com.sun.xml.ws</groupId>
            <artifactId>rt</artifactId>
            <version>4.0.1</version>
</dependency>`

File: backend/SOAPClient/src/main/java/dk/sdu/controllab/soapclient/SOAPClientApplication.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class SOAPClientApplication {
+    public static void main(String[] args) {
+        SpringApplication.run(SOAPClientApplication.class,args);
+    }
+}
```
Comment (MagnusHJensen) [inline backend/SOAPClient/src/main/java/dk/sdu/controllab/soapclient/SOAPClientApplication.java:11]: This file is not needed

File: backend/SOAPClient/pom.xml [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+                            <goals>
+                                <goal>wsdl2java</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                </plugin>
+                <plugin>
+                    <groupId>org.springframework.boot</groupId>
+                    <artifactId>spring-boot-maven-plugin</artifactId>
+                </plugin>
```
Comment (MagnusHJensen) [inline backend/SOAPClient/pom.xml:109]:
```suggestion
```

File: backend/SOAPClient/pom.xml [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
```
Comment (MagnusHJensen) [inline backend/SOAPClient/pom.xml:64]: These are not needed

File: backend/SOAPClient/src/test/java/dk/sdu/controllab/soapclient/SoapClientApplicationTests.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class SoapClientApplicationTests {
+
+    @Test
+    void contextLoads() {
+    }
+
+}
```
Comment (MagnusHJensen) [inline backend/SOAPClient/src/test/java/dk/sdu/controllab/soapclient/SoapClientApplicationTests.java:13]: This is not needed

File: backend/SOAPClient/pom.xml [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+            <version>1.5.1</version>
+        </dependency>
+    </dependencies>
+
+        <build>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.cxf</groupId>
+                    <artifactId>cxf-codegen-plugin</artifactId>
+                    <version>3.5.5</version>
```
Comment (MagnusHJensen) [inline backend/SOAPClient/pom.xml:78]:
```suggestion
                    <version>4.0.0</version>
```

File: backend/SOAPClient/pom.xml [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+        <dependency>
+            <groupId>com.sun.xml.ws</groupId>
+            <artifactId>jaxws-rt</artifactId>
+            <version>2.3.4</version>
+        </dependency>
+        <dependency>
+            <groupId>com.sun.xml.messaging.saaj</groupId>
+            <artifactId>saaj-impl</artifactId>
+            <version>1.5.1</version>
+        </dependency>
```
Comment (MagnusHJensen) [inline backend/SOAPClient/pom.xml:70]:
```suggestion
       <dependency>
            <groupId>jakarta.xml.ws</groupId>
            <artifactId>jakarta.xml.ws-api</artifactId>
            <version>4.0.0</version>
        </dependency>
        <dependency>
            <groupId>jakarta.annotation</groupId>
            <artifactId>jakarta.annotation-api</artifactId>
            <version>2.1.0</version>
        </dependency>
        <dependency>
            <groupId>com.sun.xml.ws</groupId>
            <artifactId>rt</artifactId>
            <version>4.0.1</version>
        </dependency>
```

**Review by Siole — APPROVED** [review]

**Pull request merged**